Grid/Source/Entity.php

```diff
@@ -779,4 +781,20 @@ public function getTableAlias()
     {
         return $this->tableAlias;
     }
+
+    /**
+     * @param QueryBuilder $qb
+     * @return boolean
+     */
+    protected function checkIfQueryHasFetchJoin(QueryBuilder $qb)
+    {
+        $join = $qb->getDqlPart('join');
+        foreach ($join[$this->getTableAlias()] as $join) {
+            if ($join->getJoinType() === Join::INNER_JOIN) {
```

Comment:
I'm pretty sure that join type matter. I cannot explained that, because I wrote that in February, but I'm pretty sure that it was caused by bug found by my former team's QA.

There was some kind of uncompatibility with Doctrine's paginator when there was just
`$hasJoin = !empty($this->query->getDqlPart('join'))`. I believe that `Paginator` couldn't been created successfully when `new Paginator($query, true)` constructor was called for not fetch joined query.

I'm sorry that I cannot tell you more but I haven't been working at project I was using APYDataGridBundle for 4 months.

For more details you have to ask [Pixers](https://github.com/pixers), they forked my repo and I believe there're still using APYDataGridBundle. Also their QA team had found bug I fixed here :)